### PR TITLE
PF coil builder (with `read` and `mock`)

### PR DESCRIPTION
**Pull request by CoronelBuendia**
## Description
Adds a simple builder for coilsets to enable EU-DEMO stuff to go ahead in future.

Runmodes `mock` and `read` implemented, allowing the user to simply load or specify a `CoilSet` and generate some geometry

Also cleans up plasma colouring

## Interface Changes

None

## Checklist

I confirm that I have completed the following checks:

- [x] Tests run locally and pass `pytest tests --reactor`
- [x] Code quality checks run locally and pass `flake8` and `black .`
- [x] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

Requires #521

### CODE REVIEW

File: examples/design/EU-DEMO/EUDEMO_reactor.py [hunk 1]
```diff
@@ -237,6 +237,9 @@
             "nx": 2,
             "ny": 2,
         },
+        "PF Coils": {
+            "runmode": "read",
+        },
```
Comment (DanShort12) [inline examples/design/EU-DEMO/EUDEMO_reactor.py:242]: Can we add plots for the PF coils to the example too? Or at least add them to the xz and xy plots.

**Review by DanShort12 — CHANGES_REQUESTED** [review]
Just a couple of bits. The resulting PF positioning doesn't seem quite right too, as they end up intersecting the outer leg of the TF coils (presumably something isn't being updated after the TF coil optimisation has run?).

image

File: examples/design/EU-DEMO/EUDEMO_reactor.py (hunk 1 above)
Comment (CoronelBuendia) [reply examples/design/EU-DEMO/EUDEMO_reactor.py:242]: > Just a couple of bits. The resulting PF positioning doesn't seem quite right too, as they end up intersecting the outer leg of the TF coils (presumably something isn't being updated after the TF coil optimisation has run?).
> 
> image

So here we're just reading the PF coils from the reference eqdsk, which was made without any knowledge of what the TF coils should look like. The correct PF coil positions should be delivered with the "run" mode, which I will work on in a separate PR. It's unfortunate, but this state will just leave us with a non-sensical design.

**Review by DanShort12 — APPROVED** [review]

**Comment by DanShort12** [comment]
All good from my end after the last changes - @je-cook over to you

**Comment by je-cook** [comment]
Theres a few fixes needed to the notebook but they havent come from here. Additions look really good, starting to look like a reactor properly now! Merging

**Pull request merged**